src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java

```diff
@@ -25,16 +40,19 @@
      */
     public CountryNoteAddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_COUNTRY, PREFIX_NOTE);
+                ArgumentTokenizer.tokenize(args, PREFIX_COUNTRY, PREFIX_NOTE, PREFIX_TAG);
 
         if (argMultimap.getValue(PREFIX_COUNTRY).isEmpty() || argMultimap.getValue(PREFIX_NOTE).isEmpty()
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, CountryNoteAddCommand.MESSAGE_USAGE));
         }
-        Country country = ParserUtil.parseCountry(argMultimap.getValue(PREFIX_COUNTRY).get());
+        Set<Tag> tags = tagNoteMap.getUniqueTags(argMultimap.getAllValues(PREFIX_TAG));
+
         Note note = ParserUtil.parseNote(argMultimap.getValue(PREFIX_NOTE).get());
+        Country country = ParserUtil.parseCountry(argMultimap.getValue(PREFIX_COUNTRY).get());
         CountryNote countryNote = new CountryNote(note.getNoteContent(), country);
+        countryNote.setTags(tags);
```

Comment:
tagNoteMap not being updated after setting the tags?

Reply:
Not sure about this since I followed the implementation of `client note add` command parser. @rtshkmr any comments?

Reply:
![image](https://user-images.githubusercontent.com/38996397/97428579-23e22280-1951-11eb-80bb-b767d9f4d12a.png)


the command#execute is the one doing it

Reply:
I see, can you help to check the latest commit in `CountryNoteAddCommand#execute` whether I am doing it correctly?

Reply:
@raysonkoh I think we can probably remove the model.updateTagNoteMapWithNote() and instead place it inside the function model.addCountryNote, since addClientNote is doing that as well. Then we can remove model.updateTagNoteMapWithNote() from the execute() function of both client note add and country note add, and make the method private in model

Reply:
Ok removed calls to `updateTagNoteMapWithNote` in `ClientNoteAddCommand` and `CountryNoteAddCommand`. Now the method will be called inside `ModelManager#addCountryNote` / `ModelManager#addClientNote`.